Resources/Prototypes/Entities/Structures/Storage/Tanks/tanks.yml

```diff
@@ -12,6 +12,14 @@
     state: fueltank
   - type: ReagentTank
     tankType: Fuel
+  - type: Explosive
+    # The explosive component is independent of the stored liquid :/
+    # I dream of a day where all grenades/explosives will be reagent dependent.
+    # For now let's pretend it is left over fuel vapours or something, so empty tanks can still explode.
```

Comment:
fun fact: empty tanks explode. full tanks usually just burn. (obv depends on the liquid inside, im talking the average gas for your car here)

Reply:
I realise that this sort of behaviour would complicate any eventual plans to replace explosive component with chemical-reagent based explosions, given that this becomes explosive as you have LESS reagent.

Other than just making fuel tanks a special case, I guess the way to generalise that sort of behaviour would be to either:
- Add an "oxygen" reagent to the tank as it is emptied to simulate air+fuel-vapour interactions, and allow solutions with some minimum fuel:oxygen ratio to explode when ignited
- Require atmos-chemistry with explosive fuel-vapour + air mixtures.

Reply:
first sounds fine, altough as it is rn its actually fine tbh